pkg/iac-providers/kubernetes/v1/extract-images_test.go

```diff
@@ -0,0 +1,153 @@
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.PodYAMLTemplate,
+				},
+				kind: "Pod",
+			},
+			wantContainerImageList:     []output.ContainerNameAndImage{{Name: "myapp-container", Image: "nginx"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{{Name: "myapp-container", Image: "busybox"}},
+		},
+		{
+			name: "cronjob yaml document object",
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.CronJobYAMLTemplate,
+				},
+				kind: "CronJob",
+			},
+			wantContainerImageList:     []output.ContainerNameAndImage{{Name: "hello", Image: "busybox"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{},
+		},
+		{
+			name: "job yaml document object",
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.JobYAMLTemplate,
+				},
+				kind: "Job",
+			},
+			wantContainerImageList:     []output.ContainerNameAndImage{{Name: "c", Image: "gcr.io/terrascan/job-wq-1"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{},
+		},
+		{
+			name: "deployment yaml document object",
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.DeploymentYAMLTemplate,
+				},
+				kind: "Deployment",
+			},
+			wantContainerImageList:     []output.ContainerNameAndImage{{Name: "nginx", Image: "nginx:1.14.2"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{{Name: "init", Image: "busybox"}},
+		},
+		{
+			name: "daemonset yaml document object",
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.DaemonSetYAMLTemplate,
+				},
+				kind: "DaemonSet",
+			},
+			wantContainerImageList:     []output.ContainerNameAndImage{{Name: "fluentd-elasticsearch", Image: "quay.io/fluentd_elasticsearch/fluentd:v2.5.2"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{},
+		},
+		{
+			name: "replicaset yaml document object",
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.ReplicaSetYAMLTemplate,
+				},
+				kind: "ReplicaSet",
+			},
+			wantContainerImageList:     []output.ContainerNameAndImage{{Name: "php-redis", Image: "gcr.io/google_samples/gb-frontend:v3"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{},
+		},
+		{
+			name: "replicationcontroller yaml document object",
+			args: args{
+				doc: &utils.IacDocument{
+					Type: "yaml",
+					Data: testdata.ReplicationControllerTemplate,
+				},
+				kind: "ReplicationController",
+			},
+			wantContainerImageList: []output.ContainerNameAndImage{
+				{Name: "nginx", Image: "nginx:latest"},
+				{Name: "sidecar1", Image: "sidecar-image-1"},
+				{Name: "sidecar2", Image: "sidecar-image-2"}},
+			wantInitContainerImageList: []output.ContainerNameAndImage{
+				{Name: "init1", Image: "init-image-1"},
+				{Name: "init2", Image: "init-image-2"},
+				{Name: "init3", Image: "init-image-3"}},
+		},
```

Comment:
can we add test for `StatefulSet`?

Reply:
added unit test for statefulset template.